lib/pbench/agent/base.py

```diff
@@ -65,16 +66,24 @@ def __init__(self, context):
 
         self.logger = setup_logging(debug=False, logfile=self.pbench_log)
 
-        self.ssh_opts = self.config.ssh_opts
-        self.scp_opts = self.config.scp_opts
+        ssh_opts = os.environ.get("ssh_opts")
+        if not ssh_opts:
+            ssh_opts = self.config.ssh_opts
+        self.ssh_opts = ssh_opts
+        scp_opts = os.environ.get("scp_opts")
+        if not scp_opts:
+            scp_opts = self.config.scp_opts
+        self.scp_opts = scp_opts
+
         if os.environ.get("_PBENCH_UNIT_TESTS"):
             self.date = "1900-01-01T00:00:00"
             self.date_suffix = "1900.01.01T00.00.00"
             self.hostname = "testhost"
             self.full_hostname = "testhost.example.com"
         else:
-            self.date = datetime.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%s")
-            self.date_suffix = datetime.datetime.utcnow().strftime("%Y-%m-%dT%H.%M.%s")
+            now = datetime.datetime.utcnow()
+            self.date = now.strftime("%Y-%m-%dT%H:%M:%s")
+            self.date_suffix = now.strftime("%Y.%m.%dT%H.%M.%s")
             self.hostname = socket.gethostname()
             self.full_hostname = socket.getfqdn()
```

Comment:
I don't see any [documentation](https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes) for the `%s` format specifier; moreover, when I try to use it, I get junk.  Shouldn't that be `%S`??

Also, if I might, I would suggest:
```
        if os.environ.get("_PBENCH_UNIT_TESTS"):
            self.hostname = "testhost"
            self.full_hostname = "testhost.example.com"
            now = datetime.datetime(1900, 1, 1, tzinfo=datetime.timezone.utc)
        else:
            self.hostname = socket.gethostname()
            self.full_hostname = socket.getfqdn()
            now = datetime.datetime.utcnow()

        self.date = now.strftime("%Y-%m-%dT%H:%M:%s")
        self.date_suffix = now.strftime("%Y.%m.%dT%H.%M.%s")
```
or, possibly even something like
```
        ut = os.environ.get("_PBENCH_UNIT_TESTS")
        self.hostname = "testhost" if ut else socket.gethostname()
        self.full_hostname = "testhost.example.com" if ut else socket.getfqdn()
        now = datetime.datetime.utcnow()
        if ut:
            now = datetime.datetime(1900, 1, 1, tzinfo=datetime.timezone.utc)
        self.date = now.strftime("%Y-%m-%dT%H:%M:%s")
        self.date_suffix = now.strftime("%Y.%m.%dT%H.%M.%s")
```
(The fact is, I like single assignments, and, therefore, I like ternaries, even though Black is making it hard.)

Reply:
It looks like [`%s` is](https://man7.org/linux/man-pages/man3/strftime.3.html)
```
The number of seconds since the Epoch, 1970-01-01 00:00:00 +0000 (UTC).
```
which I'm pretty sure is _not_ what we want there.

Reply:
Nice catch.